src/models/sparse.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{common::SparseEmbedding, ModelInfo};
+use ndarray::{ArrayViewD, Axis, CowArray, Dim};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SparseModel {
+    /// Splade PP Sparse vector model Qdrant/Splade_PP_en_v1
+    SPLADEPPV1,
+}
+
+pub fn models_list() -> Vec<ModelInfo<SparseModel>> {
+    vec![ModelInfo {
+        model: SparseModel::SPLADEPPV1,
+        dim: 30522,
```

Comment:
Is this field needed?
How about a new info struct for sparse models without this field?

Reply:
Ah no it's not, just felt it was cleaner to make the modelinfo generic. I could also make it 0 or wrap it in an option. What do you prefer?

Reply:
We can't change the structure since it was public. Would be breaking for any consumer.

Reply:
Does the public api change when making it generic? I couldn't run semver-checks for some reason, haven't dived into it. I made it 0 for now, but happy to create a separate struct if needed.

Reply:
I meant if we make it an `Option`.
Looks great now.